src/helpers.ts

```diff
@@ -36,6 +36,23 @@ export const EMPTY_HASH_256 = '0x00000000000000000000000000000000000000000000000
 
 export const pipelineAsync = util.promisify(stream.pipeline);
 
+// This class and enum are to throw errors that are supposed to be sent to the client
+export enum InvalidRequestErrorType {
+  invalid_hash = 'Invalid hash',
+  bad_request = 'Bad request',
+  invalid_param = 'Invalid param',
+  invalid_address = 'Invalid address',
+}
+export class InvalidRequestError extends Error {
+  type: InvalidRequestErrorType;
+  status: number;
+  constructor(msg: string, type: InvalidRequestErrorType, status: number = 400) {
+    super(msg);
+    this.type = type;
+    this.status = status;
+  }
+}
```

Comment:
We should move these (and probably the new validation functions) to the `errors.ts` file. We will need to add more useful types for debugging in the future, IMO, so we should keep them organized.